src/DevAdventCalendarCompetition/DevAdventCalendarCompetition.Repository/TestRepository.cs

```diff
@@ -25,33 +27,49 @@ public Test GetById(int testId)
             return this._dbContext.Set<Test>().FirstOrDefault(el => el.Id == testId);
         }
 
-        public void UpdateDates(Test test)
+        public Test GetByNumber(int testNumber)
         {
-            if (test is null)
-            {
-                throw new System.ArgumentNullException(nameof(test));
-            }
+            return this._dbContext.Set<Test>().FirstOrDefault(el => el.Number == testNumber);
+        }
+
+        public void AddTest(Test test)
+        {
+            this._dbContext.Set<Test>().Add(test);
+            this._dbContext.SaveChanges();
+        }
+
+        public Test GetCurrentTest()
+        {
+            return this._dbContext.Set<Test>().FirstOrDefault(el => el.Status == TestStatus.Started);
+        }
 
-            var dbTest = this._dbContext.Set<Test>().FirstOrDefault(el => el.Id == test.Id);
+        public List<Test> GetTestsWithUserAnswers()
+        {
+            return this._dbContext.Set<Test>()
+                .Include(t => t.WrongAnswers)
+                .Include(t => t.Answers)
+                .ThenInclude(ta => ta.User)
+                .OrderBy(el => el.Number).ToList();
+        }
+
+        public void UpdateDates(int testId, DateTime startDate, DateTime endDate)
+        {
+            var dbTest = this._dbContext.Set<Test>().FirstOrDefault(el => el.Id == testId);
             if (dbTest != null)
             {
-                dbTest.StartDate = test.StartDate;
-                dbTest.EndDate = test.EndDate;
+                dbTest.StartDate = startDate;
+                dbTest.EndDate = endDate;
                 this._dbContext.SaveChanges();
             }
         }
 
-        public void UpdateEndDate(Test test)
+        public void UpdateEndDate(int testId, DateTime endDate)
```

Comment:
if you are using `AddTest` and `GetTestsWithUserAnswers` then you should rename this to `UpdateEndDate`